rbi/core/object.rbi

```diff
@@ -52,6 +52,14 @@ class Object < BasicObject
   sig {returns(Integer)}
   def object_id(); end
 
+  # Returns the receiver `obj`.
+  #
+  # ```ruby
+  # obj = Object.new; obj.itself.object_id == o.object_id # => true
+  # ```
+  sig {returns(Object)}
+  def itself(); end
```

Comment:
The actual type returned at runtime would be that of the subclass of `Object` which I don't know how to express.

Also, this is my first Sorbet PR and couldn't tell if the doc comment ran an automated test.

Reply:
You can tell Sorbet that a method returns the same type of object on which the method is called with `T.self_type`:

https://sorbet.org/docs/self-type

Can you please make that change?  I will test the PR on Stripe's codebase once that change is made.

The doc comment does not run an automated test.  Your change does not require an extra test to be added; we generally consider a clean run over Stripe's codebase to be good enough for standard library rbi changes.

Reply:
Thanks for the quick response/tip. Change made.

Contributing .rbi changes is very streamlined and will continue and also encourage others who run into discrepancies.